streams/src/main/java/org/apache/kafka/streams/state/internals/SessionKeySchema.java

```diff
@@ -36,31 +36,24 @@
     private static final int SUFFIX_SIZE = 2 * TIMESTAMP_SIZE;
     private static final byte[] MIN_SUFFIX = new byte[SUFFIX_SIZE];
 
-    private String topic;
-    private final Serde<Bytes> bytesSerdes = Serdes.Bytes();
-
-    @Override
-    public void init(final String topic) {
-        this.topic = topic;
-    }
-
     @Override
     public Bytes upperRangeFixedSize(final Bytes key, final long to) {
         final Windowed<Bytes> sessionKey = new Windowed<>(key, new SessionWindow(to, Long.MAX_VALUE));
-        return Bytes.wrap(SessionKeySchema.toBinary(sessionKey, bytesSerdes.serializer(), topic));
+        return Bytes.wrap(SessionKeySchema.toBinary(sessionKey));
     }
 
     @Override
     public Bytes lowerRangeFixedSize(final Bytes key, final long from) {
         final Windowed<Bytes> sessionKey = new Windowed<>(key, new SessionWindow(0, Math.max(0, from)));
-        return Bytes.wrap(SessionKeySchema.toBinary(sessionKey, bytesSerdes.serializer(), topic));
+        return Bytes.wrap(SessionKeySchema.toBinary(sessionKey));
     }
 
     @Override
     public Bytes upperRange(final Bytes key, final long to) {
         final byte[] maxSuffix = ByteBuffer.allocate(SUFFIX_SIZE)
-            .putLong(to)
-            // start can at most be equal to end
+            // the end timestamp can be as large as possible as long as it's larger than start time
+            .putLong(Long.MAX_VALUE)
```

Comment:
That is right. Note that the names here are a bit confusing, since the semantics of the findSessions is `earliestEndTime` (named `from` here) and `latestStartTime` (named `to` here). And going back to my single-key range example in https://github.com/apache/kafka/pull/6134#issuecomment-453702517 but extending a bit to multi-key range the current code use [latestStartTime, latestStartTime] as the max-suffix to apply with key-to. This is an issue if `latestStartTime` is smaller than `earliestEndTime`: note in our aggregation it would never be the case, but for other users it is okay to call, e.g. `findSessions(100, 1)` which would use `[1,1]` as the suffix with key-to which would be missing entries.

I'll adjust the existing unit test to expose this bug as well.